cli/iconBuilder.js

```diff
@@ -0,0 +1,167 @@
+const fs = require('fs'),
+	path = require('path');
+
+const imagePath = path.join(__dirname, '../components/icons/images');
+const outputPath = path.join(__dirname, '../.generated/icons');
+
+async function getSvgsInDir(dir) {
+	return new Promise((resolve, reject) => {
+		const dirPath = path.join(imagePath, dir);
+		fs.stat(dirPath, (err, stat) => {
+			if (err) return reject(err);
+			if (!stat.isDirectory()) return resolve([]);
+			fs.readdir(dirPath, (err, files) => {
+				if (err) return reject(err);
+				const svgs = files.filter((file) => {
+					return (path.extname(file) === '.svg');
+				}).map((file) => {
+					return {category: dir, name: file.substr(0, file.length - 4)};
+				});
+				resolve(svgs);
+			});
+		});
+	});
+}
+
+async function getSvgs() {
+	return new Promise(async(resolve, reject) => {
+		fs.readdir(imagePath, async(err, files) => {
+			if (err) return reject(err);
+			let svgs = [];
+			for (let i = 0; i < files.length; i++) {
+				const newSvgs = await getSvgsInDir(files[i]);
+				svgs = svgs.concat(newSvgs);
+				if (i === files.length - 1) {
+					resolve(svgs);
+				}
+			}
+		});
+	});
+}
+
+function clearDirectoryEntry(entryPath) {
```

Comment:
I ended up writing something very similar to this.  The sync versions of the file APIs make things a little more succinct.  I hacked mine together, but here it is if you are curious: https://github.com/dbatiste/visual-diff/blob/master/filehelper.js#L21

Reply:
Haha wow yeah the sync version is a lot cleaner eh. I was tempted to just pull in `rimraf` which does it for you, but didn't want to bloat our dependencies prematurely.